44.md

```diff
@@ -0,0 +1,167 @@
+NIP-44
+======
+
+Encrypted Payloads (Versioned)
+------------------------------
+
+`optional` `author:paulmillr` `author:staab`
+
+The NIP introduces a new data format for keypair-based encryption. This NIP is versioned to allow multiple algorithm choices to exist simultaneously.
+
+The encrypted payload is encoded as base64. The first byte of the payload represents the version. The rest depends on the version.
+
+Currently defined encryption algorithms:
+
+- `0x00` - Reserved
+- `0x01` - XChaCha with same key `sha256(ecdh)` per conversation
+
+# Version 1
+
+Encoding:
+
+- `1` - 1 byte, version
+- `nonce` - 24 bytes fetched from a crytographically secure random source
+- `ciphertext` - variable number of bytes until the end
+
+`ciphertext` is created using `xchacha(key, nonce, plaintext)`
+
+Example:
+
+- Alice's private key: `5c0c523f52a5b6fad39ed2403092df8cebc36318b39383bca6c00808626fab3a`
+- Bob's private key: `4b22aa260e4acb7021e32f38a6cdf4b673c6a277755bfce287e370c924dc936d`
+
+Encrypting the message `hello` from Alice to Bob results in the following base-64 encoded payload:
+
+```
+Ac4wiXzHQ20IrriCl8ZNzPWEGre97jKP6THD4ACR
```

Comment:
This cannot be verified unless you tell us what you used for a random 24-byte nonce.

Also, I think there ought to be a test vector going in the other direction too.

Reply:
That's true, but I think it's fine just as a soft-example of what it looks like.

Reply:
I recant my statement. The nonce is in the output.